miio/gateway_scripts.py

```diff
@@ -0,0 +1,542 @@
+from json import dumps as dumps_orig
+from random import randint
+
+separators = (",", ":")
+
+
+def dumps(data):
+    return dumps_orig(data, separators=separators)
+
+
+# token in script doesn't match token of device which is used for (enc/dec)ryption
+# but they are linked somehow
+tokens = {
+    "real": "9bc7c7ce6291d3e443fd7708608b9892",
+    "encoded": "79cf21b08fb051499389f23c113477a4",
+    "data_tkn": 29576,
+}
+
+fake_device_id = "120009025"
+fake_device_model = "chuangmi.plug.v3"
+
+
+def sid_to_num(sid):
+    lumi, hex_id = sid.split(".")
+    num_id = int.from_bytes(bytes.fromhex(hex_id), byteorder="big")
+    return str(num_id)[-6:]
+
+
+action_prefix = "x.scene."
+
+# Keeping action script tail decimal int it might be used as index in some db
+action_id = {
+    "move": lambda sid: action_prefix + "1" + sid_to_num(sid),
+    "rotate": lambda sid: action_prefix + "2" + sid_to_num(sid),
+    "singlepress": lambda sid: action_prefix + "3" + sid_to_num(sid),
+    "doublepress": lambda sid: action_prefix + "4" + sid_to_num(sid),
+    "shake": lambda sid: action_prefix + "5" + sid_to_num(sid),
+    "longpress": lambda sid: action_prefix + "6" + sid_to_num(sid),
+    "flip90": lambda sid: action_prefix + "7" + sid_to_num(sid),
+    "flip180": lambda sid: action_prefix + "8" + sid_to_num(sid),
+    "shakeair": lambda sid: action_prefix + "9" + sid_to_num(sid),
+    "taptap": lambda sid: action_prefix + "10" + sid_to_num(sid),
+}
+
+
+def build_move(
+    source_sid,
+    target_ip,
+    target_model=fake_device_model,
+    target_id=fake_device_id,
+    source_model="lumi.sensor_cube.v1",
+    message_id=0,
+):
+
+    lumi, source_id = source_sid.split(".")
+    method_name = f"move_{source_id}"
+
+    move = [
+        [
+            action_id["move"](source_sid),
+            [
+                "1.0",
+                randint(1590161094, 1590162094),
+                [
+                    "0",
+                    {
+                        "did": source_sid,
+                        "extra": "[1,18,2,85,[6,256],0,0]",
```

Comment:
The 256 is specific for the move event, so make a mapping dict at the top of this file that maps events to these specific values:

extra_parameter_mapping = {"move": 256, "flip90": 64} etc.